indico/modules/events/surveys/controllers/display.py

```diff
@@ -52,8 +52,10 @@ def view_class(self):
 
 class RHSurveyList(RHSurveyBaseDisplay):
     def _process(self):
+        private_criterion = ~Survey.private | Survey.submissions.any(
```

Comment:
if it's the same query/criteria consider adding a util function to get the query/criteria.